Re-enable hdmi_cec component

newAM opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->


## Proposed change
<!--
  Describe the big picture of your changes here to communicate to the
  maintainers why we should accept this pull request. If it fixes a bug
  or resolves a feature request, be sure to link to that issue in the
  additional information section.
-->
This change re-enables the `hdmi_cec` component.
The component was removed in #37707 because it had a dependency on typing, which has since been resolved with the latest release of pyCEC, 0.4.14.

https://github.com/konikvranik/pyCEC/compare/v0.4.13...v0.4.14

## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [x] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [ ] New feature (which adds functionality to an existing integration)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #
- This PR is related to issue: 
- Link to documentation pull request: 

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [ ] The code change is tested and works locally.
- [x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] I have followed the [development checklist][dev-checklist]
- [x] The code has been formatted using Black (`black --fast homeassistant tests`)
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [x] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [x] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [ ] Untested files have been added to `.coveragerc`.

The integration reached or maintains the following [Integration Quality Scale][quality-scale]:
<!--
  The Integration Quality Scale scores an integration on the code quality
  and user experience. Each level of the quality scale consists of a list
  of requirements. We highly recommend getting your integration scored!
-->

- [ ] No score or internal
- [ ] 🥈 Silver
- [ ] 🥇 Gold
- [ ] 🏆 Platinum

<!--
  This project is very active and we have a high turnover of pull requests.

  Unfortunately, the number of incoming pull requests is higher than what our
  reviewers can review and merge so there is a long backlog of pull requests
  waiting for review. You can help here!
  
  By reviewing another pull request, you will help raise the code quality of
  that pull request and the final review will be faster. This way the general
  pace of pull request reviews will go up and your wait time will go down.
  
  When picking a pull request to review, try to choose one that hasn't yet
  been reviewed.

  Thanks for helping out!
-->

To help with the load of incoming pull requests:

- [x] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-asc+-review%3Aapproved

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[manifest-docs]: https://developers.home-assistant.io/docs/en/creating_integration_manifest.html
[quality-scale]: https://developers.home-assistant.io/docs/en/next/integration_quality_scale_index.html
[docs-repository]: https://github.com/home-assistant/home-assistant.io

homeassistant commented:
Hi @newAM, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;core#40671).

Once you do that we will be able to review and accept this pull request.

Thanks!

KevinCathcart reviewed (approved):
If you intend to help work on this integration going forward (since you are now a maintainer of the upstream library), please add yourself to the codeowners in manifest.json too. (Then run `python3 -m script.hassfest` to update the top level CODEOWNERS file).

Otherwise, this looks good, of course.

newAM commented:
> If you intend to help work on this integration going forward (since you are now a maintainer of the upstream library)

I am not sure if this is a long-term position, I only have PyPi access to the upstream right now (no repository access), I am clarifying with the owner of `pyCEC`.
I am happy to put my name down if my maintainer status of `pyCEC` becomes an ongoing role :+1:

newAM committed:
Re-enable hdmi_cec component.

newAM commented:
I have repository access now, I added myself as a codeowner.

P.S. Sorry if rebasing changes is discouraged, I did not find anything in the developer workflow about it.  I know some places dislike them, and others require them.

MartinHjelmare commented:
Please add a link to the changelog or github commit compare range for the version bump in the PR description.

newAM commented:
> Please add a link to the changelog or github commit compare range for the version bump in the PR description.

Done! It is just two small commits between 0.4.13 and 0.4.14.

Commits in 0.4.14 since 0.4.13:
* https://github.com/konikvranik/pyCEC/commit/b528dfb5f54ac8632f6cf742381dd5ee9a0a9246
* https://github.com/konikvranik/pyCEC/commit/f1cf4b636009bac97c48cc26c985a2fdd50a58b5

MartinHjelmare commented:
That's not really accurate. I've added the commit compare range to the description.

MartinHjelmare reviewed (approved):
Thanks!

newAM commented:
> That's not really accurate. I've added the commit compare range to the description.

Things got a bit out of order :/. I made the release on a branch before I realized that I did not have permissions to merge it back into maater.  The branches then got merged (rebased?) out of order.  I added the tags retroactively to preserve as much history as possible, but it is what it is.

For the future I added a changelog and have some repository access, so there will be improved accounting of changes.

TheDK commented:
Thanks newAM for driving this! I guess it will be in 0.116 next week or so?

newAM commented:
This is my first contribution to home assistant so I know as much as you do :)

It seems like releases are currently on a 3 week cycle[[1]](https://www.home-assistant.io/faq/release/) so probably next week or the week after?  There will be a beta before then though.

KevinCathcart commented:
Release is currently planned for the 7th of October. This made the cutoff
so it should be included, unless some show stopping issue is discovered
requiring it to be rolled back, which seems unlikely.

On Tue, Sep 29, 2020, 19:12 Alex <notifications@github.com> wrote:

> This is my first contribution to home assistant so I know as much as you
> do :)
>
> It seems like releases are currently on a 3 week cycle[1]
> <https://www.home-assistant.io/faq/release/> so probably next week or the
> week after? There will be a beta before then though.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/home-assistant/core/pull/40671#issuecomment-701039670>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAAWEZPLVZSQUPQDKOGR35DSIJSULANCNFSM4R3UDM2A>
> .
>
